Create and confirm intentions in one Stripe API call

reykjalin opened the pull request:
Fixes #108.

#### Changes proposed in this Pull Request

* Instead of creating and confirming intentions in 2 separate Stripe API calls, do it in the same API call. Sending 2 API calls to create and confirm intentions is done for legacy reasons, see https://github.com/Automattic/woocommerce-payments/pull/61#discussion_r297181486

#### Testing instructions

<!--
Add as many details as possible to help others reproduce the issue and test the fix.
"Before / After" screenshots can also be very helpful when the change is visual.
-->

* Make sure the response from the `create_intention` isn't an error response, and that the intention was successfully created.

Not really sure what the best way is to test this, maybe @jrodger has some more ideas?

Commit:
Create and confirm intentions in one Stripe API call

Instead of creating and confirming intentions in 2 separate Stripe API
calls, do it in the same API call. Sending 2 API calls to create and
confirm intentions is done for legacy reasons, see
https://github.com/Automattic/woocommerce-payments/pull/61#discussion_r297181486

Fixes #108.

Commit:
Add unit test for successful call to 'create_intention'

RadoslavGeorgiev reviewed (approved):
> Not really sure what the best way is to test this, maybe @jrodger has some more ideas?

The test looks good already, let's leave it like this for now. Once we abstract the Jetpack connection or do something with Jetpack DNA it will be easy to get it working.

This leaves me with one very minor nitpick. Rename the method and merge :shipit:

Commit:
Skip `create_intention` test

Skipped until the Jetpack Client dependency has been abstracted out of
the client API.

Commit:
Rename `create_intention` to `create_and_confirm_intention`

This is to reflect the change in functionality; the function now both
creates and confirms a PaymentIntent.
